testsuite/features/step_definitions/command_steps.rb

```diff
@@ -1059,45 +1059,96 @@
-  raise '/var/testsuite-data/disk-image-template.qcow2 not found' unless file_exists?(node, '/var/testsuite-data/disk-image-template.qcow2')
-  node.run("cp /var/testsuite-data/disk-image-template.qcow2 #{disk_path}")
+  case os_type
+  when "leap"
+    name = "leap-disk-image-template.qcow2"
+    net = "test-net0"
+    os = "opensuse15.4"
+  when "sles"
+    name = "sles-disk-image-template.qcow2"
+    net = "salt-sles"
+    os = "sle15sp4"
+    cloudinit = "/var/testsuite-data/cloudinit-disk.iso"
+  when "rhellike"
+    name = "rhellike-disk-image-template.qcow2"
+    net = "salt-rhellike"
+    os = "rocky8"
+  when "deblike"
+    name = "deblike-disk-image-template.qcow2"
+    net = "salt-deblike"
+    os = "ubuntu2204"
+  end
+  raise "/var/testsuite-data/#{name} not found" unless file_exists?(node, "/var/testsuite-data/#{name}")
+  raise "#{cloudinit} found" unless file_exists?(node, "#{cloudinit}")
+  node.run("cp /var/testsuite-data/#{name} #{disk_path}")
+  node.run("cp #{cloudinit} #{cloudinit_path}")
 
   # Actually define the VM, but don't start it
   raise 'not found: virt-install' unless file_exists?(node, '/usr/bin/virt-install')
-  # Use 'ide' bus for Xen and 'virtio' bus for KVM
-  bus_type = 'virtio'
   node.run(
-    "virt-install --name #{vm_name} --memory 512 --vcpus 1 --disk path=#{disk_path},bus=#{bus_type} "\
-    "--network network=test-net0 --graphics vnc,listen=0.0.0.0 "\
+    "virt-install --name #{vm_name} --memory 512 --vcpus 1 --disk path=#{disk_path},bus=virtio "\
+    "--network network=#{net} --graphics vnc,listen=0.0.0.0 "\
     "--serial file,path=/tmp/#{vm_name}.console.log "\
-    "--import --hvm --noautoconsole --noreboot --osinfo sle15sp4"
+    "--import --hvm --noautoconsole --noreboot --osinfo #{os} "\
+    "--disk path=#{cloudinit},device=cdrom,bus=sata"
   )
 end
+# rubocop:enable Metrics/BlockLength
 
+When(/^I stop the virtual machine named "([^"]*)" on "([^"]*)"$/) do |vm_name, host|
+  node = get_target(host)
+  $running, code = node.run("virsh list --name | grep #{vm_name}", check_errors: false)
+  next unless code.zero?
+  raise 'not found: virsh' unless file_exists?(node, '/usr/bin/virsh')
+  node.run("virsh destroy #{vm_name}")
+end
+
+When(/^I delete the virtual machine named "([^"]*)" on "([^"]*)"$/) do |vm_name, host|
+  node = get_target(host)
+  raise 'not found: virsh' unless file_exists?(node, '/usr/bin/virsh')
+  step %(I stop the virtual machine named "#{vm_name}" on "#{host}")
+  node.run("virsh undefine --nvram #{vm_name}")
+end
+
+# rubocop:disable Metrics/BlockLength
 When(/^I create ([^ ]*) virtual network on "([^"]*)"$/) do |net_name, host|
   node = get_target(host)
 
   networks = {
     "test-net0" => { "bridge" => "virbr0", "subnet" => 124 },
-    "test-net1" => { "bridge" => "virbr1", "subnet" => 126 }
+    "test-net1" => { "bridge" => "virbr1", "subnet" => 126 },
+    "salt-sles" => { "bridge" => "br0" },
+    "salt-rehllike" => { "bridge" => "br0" },
```

Comment:
Better `salt-rhlike` please (And not that there is also a typo in the current name)

Reply:
Jep, makes sense. Nice catch!

Reply:
Fixed.